src/core/qgsclipper.h

```diff
@@ -99,7 +105,31 @@ class CORE_EXPORT QgsClipper
 
     SIP_END
 
-    static void trimPolygon( QPolygonF &pts, const QgsRectangle &clipRect );
+    /**
+     * Trims the given polygon to a rectangular box, by modifying the given polygon in place.
+     *
+     * \param polygon polygon as 2D coordinates
+     * \param clipRect clipping rectangle
+     */
+    static void trimPolygon( QPolygonF &polygon, const QgsRectangle &clipRect );
+
+    /**
+     * Trims the given polygon to a pseudo 3D box, by modifying the given polygon in place.
+     *
+     * \param polygon polygon as 2D or 3D coordinates
+     * \param clipRect clipping 3D box
+     * \since QGIS 3.22
+     */
+    static void trimPolygon( QgsLineString &polygon, const QgsBox3d &clipRect );
```

Comment:
Can you SIP_SKIP all the new methods? The API for these should be considered temporary only -- they need reworking to use raw double lists as mentioned earlier.